cgen error: anonymous struct_init lost type name.

OS: macos, macOS, 12.5.1, 21G83
Processor: 8 cpus, 64bit, little endian, Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

getwd: /Users/shove
vmodules: /Users/shove/.vmodules
vroot: /usr/local/v
vexe: /usr/local/v/v
vexe mtime: 2022-09-08 04:13:11
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 a75a12e

Git version: git version 2.32.1 (Apple Git-133)
Git vroot status: weekly.2022.36-7-ga75a12ec
.git/config present: true
thirdparty/tcc status: thirdparty-macos-amd64 46662e20

**What did you do?**

```v
fn func(arg struct { foo string }) {
	println(arg.foo)
}

fn main() {
	func(struct { 
		foo: 'foo' 
	})
}
```

**What did you expect to see?**

No error. 

**What did you see instead?**

```
v run a.v
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/a.13017911716584131663.tmp.c:12000:13: error: expected expression
        main__func({
                   ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

C code:

```
VV_LOCAL_SYMBOL void main__func(main___VAnonStruct1 arg) {
	println(arg.foo);
}

VV_LOCAL_SYMBOL void main__main(void) {
	main__func({                                 // ==> missing: (main___VAnonStruct1)
	.foo = _SLIT("foo"),});
}

```

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -49,8 +49,7 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 		}
 	}
 	if is_anon {
-		// No name needed for anon structs, C figures it out on its own.
-		g.writeln('{')
+		g.writeln('($styp){')
 	} else if g.is_shared && !g.inside_opt_data && !g.is_arraymap_set {
 		mut shared_typ := node.typ.set_flag(.shared_f)
 		shared_styp = g.typ(shared_typ)

--- a/vlib/v/tests/anon_struct_type_test.v
+++ b/vlib/v/tests/anon_struct_type_test.v
@@ -1,0 +1,11 @@
+module main
+
+fn func(arg struct { foo string }) {
+	assert arg.foo == 'foo'
+}
+
+fn test_anon_struct_as_parameter() {
+	func(struct {
+		foo: 'foo'
+	})
+}

